src/main/java/net/dries007/tfc/common/recipes/FoodCombiningCraftingRecipe.java

```diff
@@ -0,0 +1,140 @@
+    {
+        ItemStack foodStack = ItemStack.EMPTY;
+        for (int i = 0; i < inv.getContainerSize(); i++)
+        {
+            ItemStack stack = inv.getItem(i);
+            if (!stack.isEmpty())
+            {
+                // All nonempty stacks must be a food
+                if (!stack.getCapability(FoodCapability.CAPABILITY).isPresent())
+                {
+                    return false;
+                }
+                // If there's more than one stack, keep count of the previous ones
+                if (foodStack.isEmpty())
+                {
+                    // First stack, so save and move on
+                    foodStack = stack;
+                }
+                else
+                {
+                    // Another stack, so compare. If not equal, or one is not a food, the recipe is invalid
+                    if (!FoodCapability.areStacksStackableExceptCreationDate(stack, foodStack))
+                    {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public ItemStack assemble(CraftingContainer inv)
+    {
+        ItemStack resultStack = ItemStack.EMPTY;
+        int outputAmount = 0;
+        long minCreationDate = -1;
+        for (int i = 0; i < inv.getContainerSize(); i++)
+        {
+            ItemStack stack = inv.getItem(i);
+            if (!stack.isEmpty())
+            {
+                // Get the food capability
+                IFood cap = stack.getCapability(FoodCapability.CAPABILITY).resolve().orElse(null);
+                if (cap != null)
+                {
+                    // Increment output amount
+                    outputAmount++;
+
+                    // Find the min creation date
+                    if (minCreationDate == -1 || minCreationDate > cap.getCreationDate())
+                    {
+                        minCreationDate = cap.getCreationDate();
+                    }
+
+                    // And save the stack
+                    if (resultStack.isEmpty())
+                    {
+                        resultStack = stack.copy();
+                    }
+                }
+            }
+        }
+
+        // Update the capability and count of the result
+        resultStack.setCount(outputAmount);
+        final long date = minCreationDate;
+        resultStack.getCapability(FoodCapability.CAPABILITY).ifPresent(cap -> cap.setCreationDate(date));
+        return resultStack;
+    }
+
+    @Override
+    public ItemStack getResultItem()
+    {
+        return ItemStack.EMPTY;
+    }
+
+    @Override
+    public ResourceLocation getId()
+    {
+        return id;
+    }
+
+    @Override
+    public RecipeSerializer<?> getSerializer()
+    {
+        return TFCRecipeSerializers.FOOD_COMBINING_CRAFTING.get();
+    }
+
+    public static class Serializer extends RecipeSerializerImpl<FoodCombiningCraftingRecipe>
```

Comment:
Instead of this class, register `() -> new SimpleRecipeSerializer(FoodCombiningCraftingRecipe::new)`